### PR TITLE
feat: add required indicator in jobs and quiz

### PR DESCRIPTION
**Pull request by iamejaaz**
## Reason of PR
- Add the required indicator to the Jobs and Quiz forms for fields that are missing.

## Screenshot
image

### CODE REVIEW

File: frontend/src/components/Modals/Question.vue [hunk 1]
```diff
@@ -56,12 +56,14 @@
...
 						:options="['Choices', 'User Input', 'Open Ended']"
 						class="pb-2"
+						:required="true"
 					/>
 					<div v-if="question.type == 'Choices'" class="divide-y border-t">
 						<div v-for="n in 4" class="space-y-4 py-2">
 							<FormControl
 								:label="__('Option') + ' ' + n"
 								v-model="question[`option_${n}`]"
+								:required="n == 1 ? true : false"
```
Comment (pateljannat) [inline frontend/src/components/Modals/Question.vue:66]: minimum 2 options are required for Choices

**Review by pateljannat — CHANGES_REQUESTED** [review]

File: frontend/src/components/Modals/Question.vue (hunk 1 above)
Comment (iamejaaz) [reply frontend/src/components/Modals/Question.vue:66]: Hey @pateljannat Please check the video—only one option is required. 🤔 ??

https://github.com/user-attachments/assets/6b677a99-dc9d-4cca-b828-51a64ca0fcdb

Comment (iamejaaz) [reply frontend/src/components/Modals/Question.vue:66]: Update: The form has been saved without options. Is this an issue? If so, should I fix it in this PR?

https://github.com/user-attachments/assets/24c29d89-fda7-4eab-85e9-cc3d256500a2

Comment (pateljannat) [reply frontend/src/components/Modals/Question.vue:66]: > Hey @pateljannat Please check the video—only one option is required. 🤔 ??

If you check the LMS Question doctype, option_2 field has a mandatory_depends_on condition. This was getting triggered properly from the desk, but not working for the portal or when the question was created through API. 

Added a validation to fix the same https://github.com/frappe/lms/pull/1133

Please add required icon for option_2 if type is Choices.

Comment (iamejaaz) [reply frontend/src/components/Modals/Question.vue]: Done
image

File: frontend/src/components/Modals/Question.vue [hunk 2]
```diff
@@ -82,6 +84,7 @@
 						<FormControl
 							:label="__('Possibility') + ' ' + n"
 							v-model="question[`possibility_${n}`]"
+							:required="n <= 2 ? true : false"
```
Comment (pateljannat) [inline frontend/src/components/Modals/Question.vue:87]: For possibilities, only 1 is enough

**Review by pateljannat — CHANGES_REQUESTED** [review]

File: frontend/src/components/Modals/Question.vue (hunk 2 above)
Comment (iamejaaz) [reply frontend/src/components/Modals/Question.vue]: Done

**Pull request merged**